user/printf.c

```diff
@@ -259,26 +251,139 @@ static inline char getc(void)
 // Function to getline from user input
 void getlines(char *restrict buffer, size_t length)
```

Comment:
I feel like this getlines function could be simplified by adding modularity. This would help with improving readability. This is okay for now, but I will probably refactor it in the future.